pkgs/development/libraries/openssl/default.nix

```diff
@@ -58,6 +59,9 @@ stdenv.mkDerivation rec {
 
     # remove dependency on Perl at runtime
     rm -r $out/etc/ssl/misc $out/bin/c_rehash
+
+    # configure the default trust store
+    ${optionalString (defaultCertificate != null) "ln -s ${defaultCertificate} $out/etc/ssl/cert.pem"}
```

Comment:
Hm, how does it create more impurity, compared to the symlink?

Regarding Darwin, there is no good default value, so `/etc/ssl/certs/ca-certificates.crt` is as good as any :-)